### ISSUE TITLE
Unable to persist optional enum fields using contrib-postgres

### ISSUE THREAD

**Issue by onurzdg** [tpolecat/doobie#365]
It is not possible to persist optional enum fields using pgEnumString from doobie.contrib.postgresql. However, I found that the "pgOptEnumString" method below works.

```
 // same as the one in the library
   private def enumPartialMeta(name: String): Meta[String] =
     Meta.basic1[String](jdbctype.Other, Nil,
       (rs, n) => rs.getString(n),
       (n, a) => FPS.setObject(n, new PGobject <| (_.setValue(a.toString)) <| (_.setType(name))),
       (n, a) => FRS.updateObject(n, new PGobject <| (_.setValue(a.toString)) <| (_.setType(name))))

 // works
  def pgOptEnumString[A](name: String, f: Option[String] => A, g: A => Option[String]): Atom[A] =
    Atom.fromScalaTypeOption(enumPartialMeta(name)).xmap(f,g)
```

**Comment by tpolecat** [comment]
Finally looking at this and I don't quite get it. How does this permit you to write `None`?

**Comment by tpolecat** [comment]
This was fixed in #515
